torch/distributed/_shard/sharding_spec/chunk_sharding_spec.py

```diff
@@ -22,6 +23,8 @@
     # from run-time to resolve circular dependency.
     from torch.distributed._shard.sharded_tensor import ShardedTensor
 
+_ShardingDim: TypeAlias = Union[int, str]
```

Comment:
I'm guess something imported the type Alias from the dataclass internally? I mean we can keep it as a class based TypeAlias but seems silly

Reply:
The error is:
```
Incompatible parameter type [6]: In call `torch.distributed._shard.sharding_spec.chunk_sharding_spec.ChunkShardingSpec.__init__`, for argument `placements`, expected `List[Union[str, _remote_device]]` but got `List[str]`.
```

Reply:
So I guess we need to fix the wrong type internally and not much we can do here?